### PR TITLE
[wlroots] Add new port

### PR DESCRIPTION
**Pull request by handicraftsman**
Added wlroots port to vcpkg.

Only Linux-based triplets are supported because wlroots depends on libraries native to it.

I did not update ci-baseline yet, will do in next commit.

Please note that I patched meson build scripts to let them find vcpkg-provided packages (only pixman at the moment).

Also note that it depends on some system libraries like libdrm or x11 libraries (not sure yet, need to recheck configuration info).

### CODE REVIEW

**Comment by handicraftsman** [comment]
Sorry for all this mess, I should read nearby code more carefully from now.

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1950,6 +1950,15 @@ wintoast:arm-uwp=fail
 wintoast:x64-linux=fail
 wintoast:x64-osx=fail
 wintoast:x64-uwp=fail
+wlroots:arm64-windows=skip
+wlroots:arm-uwp=skip
+# depends on native drm and x11 libraries
+wlroots:x64-linux=fail
```
Comment (PhoebeHui) [inline scripts/ci.baseline.txt:1956]: If it only supports linux, could you remove 'wlroots:x64-linux=fail', and could please give message to user?
eg: https://github.com/microsoft/vcpkg/blob/d10918a71ab43df46689f52fa3280830ae79ffee/ports/numactl/portfile.cmake#L15

I will ask team member to update VM, then we can test it via CI pipeline.

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
Thanks for your PR!

**Comment by handicraftsman** [comment]
Okay, will do, thanks for response.

File: ports/wlroots/CONTROL [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Source: wlroots
+Version: 0.10.1
+Description: A modular Wayland compositor library
+Build-Depends: tool-meson, pixman
```
[the remark this answers is not in the document]
Comment (handicraftsman) [reply ports/wlroots/CONTROL:4]: By the way, update this please, these fields were not mentioned in the docs: https://vcpkg.readthedocs.io/en/latest/maintainers/control-files/

**Comment by PhoebeHui** [comment]
@BillyONeal, wlroots requires more libraries from the system, currently only linux VM required to install them, what's process for this now? could you help confirm?

Does the author need to update https://github.com/microsoft/vcpkg/blob/master/scripts/azure-pipelines/linux/provision-image.sh?

message("wlroots currently requires the following libraries from the system package manager:
    wayland
    wayland-protocols
    EGL
    GLESv2
    libdrm
    GBM
    libinput
    xkbcommon
    udev
    pixman
    systemd (optional, for logind support)
    elogind (optional, for logind support on systems without systemd)
If you choose to enable X11 support:
    
    xcb
    xcb-composite
    xcb-xfixes
    xcb-xinput
    xcb-image
    xcb-render
    x11-xcb
    xcb-errors (optional, for improved error reporting)
    x11-icccm (optional, for improved Xwayland introspection)")

**Comment by BillyONeal** [comment]
Yes, the author would need to edit the provisioning script to install the additional packages necessary, and then we would tag this PR "depends:vm-update". Then next time someone updates the image used for the Linux VMs they would make sure to collect all such changes to that script and do a deploy (example #11365).

**Comment by PhoebeHui** [comment]
@BillyONeal, thanks for confirm this!

@handicraftsman, could you please also update scripts/azure-pipelines/linux/provision-image.sh?

**Comment by LilyWangL** [comment]
Pinging @handicraftsman for response, could you please also update scripts/azure-pipelines/linux/provision-image.sh?

**Comment by handicraftsman** [comment]
There has been a new release of wlroots, so i am closing this (temporarily i think).

**Pull request closed without merging**